src/components/filters/FilterModal.jsx

```diff
@@ -1,44 +1,79 @@
 import "./filtermodal.css";
 export function FilterModal({ setShowFilterModal }) {
-  const labels = ["label1", "label2", "label3", "label4"];
+  const { tags } = useNotes();
+  const { stateFilter, dispatchFilter } = useFilterNotes();
+  const { priority, sortByDate, tags: filterTags } = stateFilter;
   return (
     <>
       <div className="d-flex modal-container">
         <div className="modal children-stacked">
           <>
-            <div className="modal-header d-flex">
-              <h5>Filter by:</h5>
-            </div>
+            <h5 className="text-center">Filter by:</h5>
             <div className="modal-body">
               <h6>Tags:</h6>
               <ul className="d-flex grid-gap filter-tags">
-                {labels &&
-                  labels?.map(item => {
-                    return (
-                      <li className="d-flex">
-                        <span className="">
-                          <label>
-                            <input type="checkbox" /> &nbsp;
-                            {item}
-                          </label>
-                        </span>
-                      </li>
-                    );
-                  })}
+                {tags?.map((item, idx) => {
+                  return (
+                    <li className="d-flex" key={"filter-tag" + idx}>
+                      <span className="">
+                        <label>
+                          <input
+                            type="checkbox"
+                            id={item}
+                            name={item}
+                            onChange={() => {
+                              dispatchFilter({
+                                type: "SET_TAG",
+                                payload: item,
+                              });
+                            }}
+                            checked={
+                              filterTags.find(t => t === item) ? true : false
+                            }
+                            value={item}
+                          />
+                          {item}
+                        </label>
+                      </span>
+                    </li>
+                  );
+                })}
               </ul>
               <hr></hr>
 
               <h6>Date:</h6>
               <ul className="list list-no-bullet d-flex grid-gap">
                 <li>
                   <label htmlFor="oldest">
-                    <input type="radio" id="oldest" name="oldest" />
+                    <input
+                      type="radio"
+                      id="oldest"
+                      name="oldest"
+                      checked={sortByDate === "oldest"}
+                      value="oldest"
+                      onChange={() =>
+                        dispatchFilter({
+                          type: "SET_DATE",
+                          payload: "oldest",
+                        })
+                      }
```

Comment:
Apart from the payload value the onChange function seems repetitive so you can create a function named as dateFilter that takes the value of the payload and calls the filter dispatch function. For demonstration you can understand it through a short snippet of mine.
```
//function definition
const applyDateFilter=(valueGiven)=>{
dispatchFilter({type:"SET_DATE",payload:valueGiven})
}
//calling function within Input
onChange={(e)=>{
applyDateFilter(e.target.value)
}}
```
Since you have already defined a value attribute within inputSections so for onChange you can call this function using it.

Can follow this logic for latest one too
And similarly for sorting notes according to priority you can create a priority filter function that takes the attribute to be applied as we did in the applyDateFilter function

Reply:
Thanks for this Rohsan, I have changed it to use e.target.value